javascript/javascript3/week3/shoping_cart.js

```diff
@@ -0,0 +1,134 @@
+console.log('Shoping cart');
+
+                /* Part 1 */
+
+     class Product {
+     constructor(name, price) {
+      this.name = name;
+      this.price = price;
+     }
+
+     async convertToCurrency(curr) {
+        const currency=curr.toUpperCase();
+        let price = 0;
+        const rice = await fetch("https://api.exchangeratesapi.io/latest?base=DKK")
+          .then((promise) => promise.json())
+          .then((data) => {
+            for (let item in data.rates) {
+              if (item == currency) {
+                const currentrate = data.rates[item];
+                price = this.price * currentrate.toFixed(2);
+              }
+            }
+          });
+        return 'Price of '+this.price+' DKK is : '+ price+' '+currency;
+      }
+    }
+  
+  
+     class ShoppingCart {
+     constructor() {
+      this.products = [];
+      //console.log(this.products)
+    }
+  
+    addProduct(product) {
+       // Implement functionality here
+        this.products.push(product);
+        console.log('added product');
+        console.log(product.name);
+      
+    }
+  
+    removeProduct(product) {
+        // Implement functionality here
+        for(let i=0;i<this.products.length;i++){
+            if(this.products[i].name==product.name){
+            this.products.splice(i, 1);
+            console.log('\n\n'+product.name +' is removed  ')
+        }
+        }
+    }
+  
+    searchProduct(productName) {
+      // Implement functionality here
+      this.products.forEach(oneProduct=>{
+        if(oneProduct.name==productName){
+        console.log('\n\n Searched product');
```

Comment:
console.log is just for you to see, usually you want a function to return some result. Then you can console.log after you return from the function.

Reply:
Done
 let serchedProduct= this.products.filter(oneProduct => oneProduct.name == productName) ;
    
      console.log("\n\n Searched product");
      console.log(serchedProduct[0].name);